File: OP_labs_cs/lab5_cs/lab5_cs/Triad.cs
namespace lab5_cs
{
    public abstract class Triad
    {
        public int First;
        public int Second;
        public int Third;
        
        protected Triad(int first, int second, int third)
        {
            First = first;
            Second = second;
            Third = third;
        }

        public abstract void IncreaseFirst(int num);
        public abstract void IncreaseSecond(int num);
        public abstract void IncreaseThird(int num);

        public abstract void ReduceFirst(int num);
        public abstract void ReduceSecond(int num);
        public abstract void ReduceThird(int num);
    }
}

File: OP_labs_py/lab5_py/Triad.py
from abc import *


class Triad:
    def __init__(self, first, second, third):
        self.first = first
        self.second = second
        self.third = third

    @abstractmethod
    def increase_first(self, num):
        pass

    @abstractmethod
    def increase_second(self, num):
        pass

    @abstractmethod
    def increase_third(self, num):
        pass

    @abstractmethod
    def reduce_first(self, num):
        pass

    @abstractmethod
    def reduce_second(self, num):
        pass

    @abstractmethod
    def reduce_third(self, num):
        pass
